kosat-core/src/commonMain/kotlin/org/kosat/cnf/CNF.kt

```diff
@@ -1,12 +1,29 @@
 package org.kosat.cnf
 
+import okio.Buffer
 import okio.BufferedSource
 import kotlin.math.abs
 
-class CNF(
+data class CNF(
     val clauses: List<List<Int>>,
     val numVars: Int = determineNumberOfVariables(clauses),
 ) {
+    fun toString(includeHeader: Boolean): String {
```

Comment:
This should be called "`toDimacsString()`" / "`toDimacs()`" or something like that. See `Lit::toDimacs`

Reply:
Also, consider replacing this method with `writeDimacs(sink: BufferedSink)`